src/main/app/controller/courts/PostcodesController.ts

```diff
@@ -0,0 +1,220 @@
+    const areasOfLaw = req.body.areasOfLaw?.split(',') ?? [];
+    if (!CSRF.verify(req.body.csrfToken)) {
+      return this.get(req, res, '', existingPostcodes, this.addErrorMsg, areasOfLaw, courtTypes);
+    }
+
+    // If we have no postcodes selected
+    const postcodesToDelete: string[] = req.body.selectedPostcodes ?? [];
+    if (!postcodesToDelete.length) {
+      return this.get(req, res, '', existingPostcodes, this.noSelectedPostcodeMsg, areasOfLaw, courtTypes);
+    }
+
+    // Send the postcodes to fact-api to delete them
+    await req.scope.cradle.api.deletePostcodes(req.params.slug, postcodesToDelete)
+      .then(async () => {
+        // Remove the values from the existing list and return
+        const existingMinusDeleted =
+          existingPostcodes.filter( ( postcode ) => !postcodesToDelete.includes( postcode ) );
+        await this.get(req, res, '', existingMinusDeleted, '', areasOfLaw, courtTypes, true);
+      })
+      .catch(async () =>
+        await this.get(req, res, '', existingPostcodes,
+          this.deleteErrorMsg + postcodesToDelete, areasOfLaw, courtTypes));
+  }
+
+  public async put(
+    req: AuthedRequest,
+    res: Response): Promise<void> {
+
+    const existingPostcodes: string[] = req.body.existingPostcodes?.split(',') ?? [];
+    const courtTypes = req.body.courtTypes?.split(',') ?? [];
+    const areasOfLaw = req.body.areasOfLaw?.split(',') ?? [];
+    if (!CSRF.verify(req.body.csrfToken)) {
+      return this.get(req, res, '', existingPostcodes, this.moveErrorMsg, areasOfLaw, courtTypes);
+    }
+
+    // If we have no postcodes selected
+    const postcodesToMove: string[] = req.body.selectedPostcodes ?? [];
+    if (!postcodesToMove.length || !req.body.selectedCourt) {
+      return this.get(req, res, '', existingPostcodes,
+        this.noSelectedPostcodeOrCourtMsg, areasOfLaw, courtTypes);
+    }
+
+    // Send the postcodes to fact-api to delete them
+    await req.scope.cradle.api.movePostcodes(req.params.slug, req.body.selectedCourt, postcodesToMove)
+      .then(async () => {
+        // Remove the values from the existing list and return
+        const existingMinusDeleted =
+          existingPostcodes.filter( ( postcode ) => !postcodesToMove.includes( postcode ) );
+        await this.get(req, res, '', existingMinusDeleted, '', areasOfLaw, courtTypes, true);
+      })
+      .catch(async (reason: AxiosError) => {
+        // conflict, postcode(s) already exists on the database
+        if(reason.response?.status === 409) { // conflict, postcode already exists on the destination court
+          await this.get(req, res, '', existingPostcodes,
+            this.moveErrorDuplicatedMsg + postcodesToMove, areasOfLaw, courtTypes);
+        }
+        else {
+          await this.get(req, res, '', existingPostcodes,
+            this.moveErrorMsg + postcodesToMove, areasOfLaw, courtTypes);
+        }
+      });
+  }
+
+  private checkCountyAreasOfLaw(courtAreasOfLaw: string[]): string[] {
```

Comment:
Minor point - the naming suggests it's checking something but it's really filtering out certain types of law.